python/mxnet/gluon/rnn/rnn_layer.py

```diff
@@ -291,8 +291,16 @@ def __init__(self, hidden_size, num_layers=1, activation='relu',
                                   'rnn_'+activation, **kwargs)
 
     def state_info(self, batch_size=0):
-        return [{'shape': (self._num_layers * self._dir, batch_size, self._hidden_size),
-                 '__layout__': 'LNC'}]
```

Comment:
I will revert this part. This is indeed not the right place to address the layout issue. Sorry for the confusion.